### ISSUE TITLE
Dashboard still visible when using basicauth docs configuration

### ISSUE THREAD

**Issue by pacoferre** [containous/traefik#5561]
### Do you want to request a *feature* or report a *bug*?

Bug

### Did you try using a 1.7.x configuration for the version 2.0?

- [ ] Yes
- [X] No

### What did you do?

Set up a new host with docker + traefik v2.
Set configuration as described in https://docs.traefik.io/v2.0/operations/api/#configuration
![image](https://user-images.githubusercontent.com/10376690/65978310-c476c200-e473-11e9-8ecc-7db53a911b5b.png)

### What did you expect to see?

Asked for Username / password for any url ...server:8080/api/... or ...server:8080/dashboard/...

### What did you see instead?

Only Username / password asked for URLs ...server:8080/api and ...server:8080/dashboard

![image](https://user-images.githubusercontent.com/10376690/65979982-9050d080-e476-11e9-9df3-565ef0d75fbf.png)

If I press Cancel, all other URLs works without any password asked, like this...
...server:8080/dashboard/#/
...server:8080/api/http/middlewares

.../dashboard/#/http/routers/api@docker  shows all info
![image](https://user-images.githubusercontent.com/10376690/65979072-ffc5c080-e474-11e9-8bb9-1782278375a9.png)

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
Version:      2.0.1
Codename:     montdor
Go version:   go1.13.1
Built:        2019-09-26T16:18:03Z
OS/Arch:      linux/amd64
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

docker-compose.yml
```yml
version: '3'

services:
  reverse-proxy:
    image: traefik:v2.0
    command: --providers.docker --api.insecure=true
    ports:
      - "80:80"
      - "8080:8080"
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
    labels:
      - "traefik.http.routers.api.rule=PathPrefix(`/api`) || PathPrefix(`/dashboard`)"
      - "traefik.http.routers.api.service=api@internal"
      - "traefik.http.routers.api.middlewares=auth"
      - "traefik.http.middlewares.auth.basicauth.users=USER:PASSW"
```

No other services running on docker.

**Comment by dduportal** [comment]
Hi @pacoferre , thanks for your interest or for the clear report!

Could you check this pull request on the documentation: #5523 , where we try to clarify the possible setups.

In your case, you're mixing both "secure" and "insecure" configuration which has unexpected behaviors.

Steps by steps:

- Remove `--api.insecure=true` flag (and port `8080` won't be used anymore)
- Add `--api.dashboard=true` to explicitly enable the dashboard, available on the "Traefik Service" named `api@internal` along with the API

The you can reach the dashboard at `http://server/dashboard/` (trailing slash is mandatory).

**Comment by pacoferre** [comment]
> Hi @pacoferre , thanks for your interest or for the clear report!
> 
> Could you check this pull request on the documentation: #5523 , where we try to clarify the possible setups.
> 
> In your case, you're mixing both "secure" and "insecure" configuration which has unexpected behaviors.
> 
> Steps by steps:
> 
> * Remove `--api.insecure=true` flag (and port `8080` won't be used anymore)
> * Add `--api.dashboard=true` to explicitly enable the dashboard, available on the "Traefik Service" named `api@internal` along with the API
> 
> The you can reach the dashboard at `http://server/dashboard/` (trailing slash is mandatory).

Hi @dduportal, thanks a lot for your reply.

I did what you told me and everything is working now :)

Best regards,

Paco

**Comment by traefiker** [comment]
Closed by #5523.

**Comment by gregfr** [comment]
Can you please elaborate on how you made it work? I copied and pasted the same setup and it's not working properly:
- `/dashboard/` gives me an "empty" dashboard (only the titles, with gray content)
- and redirects to `/dashboard/#/` which gives a ERR_CONNECTION_REFUSED

I'm lost, how can I add basic auth to the dashboard?

Thanks in advance
